features/nanostack/nanostack-hal-mbed-cmsis-rtos/arm_hal_fhss_timer.cpp

```diff
@@ -53,45 +54,41 @@ static EventQueue *equeue;
 // an initialized-data cost.
 struct fhss_timeout_s {
     void (*fhss_timer_callback)(const fhss_api_t *fhss_api, uint16_t) = nullptr;
-    uint32_t start_time = 0;
-    uint32_t stop_time = 0;
     bool active = false;
     SingletonPtr<Timeout> timeout;
 };
 
 fhss_timeout_s fhss_timeout[NUMBER_OF_SIMULTANEOUS_TIMEOUTS];
 
-static uint32_t read_current_time(void)
-{
-    return timer->read_us();
-}
-
 static fhss_timeout_s *find_timeout(void (*callback)(const fhss_api_t *api, uint16_t))
 {
-    for (int i = 0; i < NUMBER_OF_SIMULTANEOUS_TIMEOUTS; i++) {
-        if (fhss_timeout[i].fhss_timer_callback == callback) {
-            return &fhss_timeout[i];
+    for (fhss_timeout_s &t : fhss_timeout) {
+        if (t.fhss_timer_callback == callback) {
+            return &t;
         }
     }
-    return NULL;
+    return nullptr;
 }
 
 static fhss_timeout_s *allocate_timeout(void)
 {
-    for (int i = 0; i < NUMBER_OF_SIMULTANEOUS_TIMEOUTS; i++) {
-        if (fhss_timeout[i].fhss_timer_callback == NULL) {
-            return &fhss_timeout[i];
+    for (fhss_timeout_s &t : fhss_timeout) {
+        if (t.fhss_timer_callback == NULL) {
+            return &t;
         }
     }
-    return NULL;
+    return nullptr;
 }
 
 static void fhss_timeout_handler(void)
 {
-    for (int i = 0; i < NUMBER_OF_SIMULTANEOUS_TIMEOUTS; i++) {
-        if (fhss_timeout[i].active && ((fhss_timeout[i].stop_time - fhss_timeout[i].start_time) <= (read_current_time() - fhss_timeout[i].start_time))) {
-            fhss_timeout[i].active = false;
-            fhss_timeout[i].fhss_timer_callback(fhss_active_handle, read_current_time() - fhss_timeout[i].stop_time);
+    for (fhss_timeout_s &t : fhss_timeout) {
+        if (t.active) {
+            microseconds remaining_time = t.timeout->remaining_time();
+            if (remaining_time <= 0s) {
+                t.active = false;
+                t.fhss_timer_callback(fhss_active_handle, remaining_time.count());
```

Comment:
Should this be passing `-remaining_time.count()`?

Reply:
Okay, tested it now and that is correct. Passing this negative fixed it.